daemon/libnetwork/portallocator/setsockopt_linux_386.go

```diff
@@ -0,0 +1,16 @@
+package portallocator
```

Comment:
Is this only `i386` or also `armhf` (arm 32-bit)?

Reply:
I think Go never added support for `setsockopt` on i386 because it's not available on older versions of Linux. `man 2 socketcall` says:

       On x86-32, socketcall() was historically the only entry point for
       the sockets API.  However, starting in Linux 4.3, direct system
       calls are provided on x86-32 for the sockets API.

However, armhf supports it: https://github.com/golang/go/blob/691af6ca28dad9c72e51346fe10c6aaadc3f940b/src/syscall/zsysnum_linux_arm.go#L93

I'll add a comment to mention that.